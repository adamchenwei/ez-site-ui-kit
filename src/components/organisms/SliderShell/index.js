import React from 'react';
import PropTypes, { oneOfType } from 'prop-types';
import Container from './style/Container';

const SliderShell = (props) => {
  const {
    className,
    style,
    children,
    slides,
    currentSlideIndex,
  } = props;

  function showSlides() {
    return slides.map((slide, index) => {
      const Slide = slide.component;
      const slideProps = slide.props;
      const zIndexLevel = 890 + index;
      return (
        <Slide
          {...slideProps}
          key={slide.id || index}
          style={{
            zIndex: zIndexLevel,
            opacity: index === currentSlideIndex ? 1 : 0,
            transition: 'opacity 500ms linear',
          }}
        />);
    });
    // const ASlide = slides[0];
    // return <h1> <ASlide /></h1>;
  }
  return (
    <Container style={style} className={className} >
      {children || showSlides()}
    </Container>
  );
};

SliderShell.propTypes = {
  currentSlideIndex: PropTypes.number,
  style: PropTypes.objectOf(oneOfType([
    PropTypes.string,
    PropTypes.number,
  ])),
  className: PropTypes.string,
  children: PropTypes.node,
  slides: PropTypes.arrayOf(PropTypes.any),
};

SliderShell.defaultProps = {
  currentSlideIndex: 0,
  style: {},
  className: '',
  children: null,
  slides: [],
};

export default SliderShell;
