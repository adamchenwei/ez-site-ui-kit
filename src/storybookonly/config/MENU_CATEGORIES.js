// Atomic Design Component Structure
export const COMPONENT_LABEL = 'Component';
export const ATOM = 'Component';
export const MOLECULE = 'Molecule';
export const ATOM_LABEL = 'Atom';
export const MOLECULE_LABEL = 'Molecule';
export const ORGANISM = 'Organism';
export const ORGANISM_LABEL = 'Organism';
export const TEMPLATE = 'Template';
export const PAGE = 'Page';
// Framework Base Options
export const BOOTSTRAP = 'Bootstrap';
export const MATERIALUI = 'Material-ui';
export const ES6 = 'ES6';

// Containers for Redux or Global State
export const CONTAINER = 'Container';

export default {
  COMPONENT_LABEL,
  ATOM,
  MOLECULE,
  ATOM_LABEL,
  MOLECULE_LABEL,
  ORGANISM,
  ORGANISM_LABEL,
  TEMPLATE,
  PAGE,
  BOOTSTRAP,
  MATERIALUI,
  ES6,
  CONTAINER,
};
