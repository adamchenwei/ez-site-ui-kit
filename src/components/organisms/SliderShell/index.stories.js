import React from 'react';
import { storiesOf } from '@storybook/react';
import SliderShell from './index';
import { COMPONENT_LABEL, ES6, ORGANISM_LABEL } from '../../../storybookonly/config/MENU_CATEGORIES';
import SlideBasicCallToAction from './../../molecules/SlideBasicCallToAction';
// import PositionedSlideContainer from '../PositionedSideShell/style/PositionedSlideContainer';
import PositionedSlideShell from '../PositionedSideShell';
import {slides} from './__mocks__/slides'; //eslint-disable-line

const imageSourcePath1 = './placeholder/images/1040x700color515151.svg';
const imageSourcePath2 = './placeholder/images/1040x700color355952.svg';
const imageSourcePath3 = './placeholder/images/1040x700color373e7a.svg';


// class StateHolder extends Component {
//   constructor(props) {
//     this.state = {
//       slidesOrderedIdsList: props.slidesOrderedIdsList,
//     }
//   }

//   render() {
//     return (
//       <SliderShell>

//       </SliderShell>
//     )
//   }
// }

// const PositionedSlide = props => (
//   <PositionedSlideContainer>
//     <SlideBasicCallToAction {...props} />
//   </PositionedSlideContainer>
// );

// TODO - funny stuff, the same slides in slides file is making the slides disappear!!!!
// const SLIDES = slides;
const SLIDES = [
  {
    component: PositionedSlideShell,
    props: {
      title: 'Happy',
      subTitle: 'View Our Happy Slide',
      paragraph: 'we got this beautiful slide that is crazy good, please try it out yourself when you have time!',
      callToActionButtonConfig: {
        title: 'Click Me',
        link: 'https://www.google.com',
        customButtonComponent: () => <button>my custom button 1</button>,
        customButtonComponentProps: {},
      },
      imageSourcePath: imageSourcePath2,
    },
  },
  {
    component: PositionedSlideShell,
    props: {
      title: 'Enjoyable',
      subTitle: 'View Our Enjoyable Slide',
      paragraph: 'we got this beautiful slide that is crazy good, please try it out yourself when you have time!',
      callToActionButtonConfig: {
        title: 'Click Me',
        link: 'https://www.google.com',
        customButtonComponent: () => <button>my custom button 1</button>,
        customButtonComponentProps: {},
      },
      imageSourcePath: imageSourcePath1,
    },
  },
  {
    component: PositionedSlideShell,
    props: {
      title: 'Funny',
      subTitle: 'View Our Funny Slide',
      paragraph: 'we got this beautiful slide that is crazy good, please try it out yourself when you have time!',
      callToActionButtonConfig: {
        title: 'Click Me',
        link: 'https://www.google.com',
        customButtonComponent: () => <button>my custom button 1</button>,
        customButtonComponentProps: {},
      },
      imageSourcePath: imageSourcePath3,
    },
  },
];


storiesOf(`${COMPONENT_LABEL}/${ES6}/${ORGANISM_LABEL}/SliderShell/Stand Alone`, module)
  .addDecorator(story => (
    <div style={{ marginTop: '50px' }}>
      <section style={{ width: '100%', height: '450px' }}>
        {story()}
      </section>
    </div>
  ))
  .add('default', () => (
    <SliderShell>
      <SlideBasicCallToAction
        title="Enjoyable"
        subTitle="View Our Beautiful Slide"
        paragraph="we got this beautiful slide that is crazy good, please try it out yourself when you have time!"
        callToActionButtonConfig={{
          title: 'Click Me',
          link: 'https://www.google.com',
          customButtonComponent: () => <button>my custom button</button>,
          customButtonComponentProps: {},
        }}
        imageSourcePath={imageSourcePath1}
      />
    </SliderShell>
  ))
  .add('3 slides: 1st slide / default', () => (
    <SliderShell
      slides={SLIDES}
    />
  ))
  .add('3 slides: 2nd slide', () => (
    <SliderShell
      slides={SLIDES}
      currentSlideIndex={1}
    />
  ))
  .add('3 slides: last slide', () => (
    <SliderShell
      slides={SLIDES}
      currentSlideIndex={2}
    />
  ));
