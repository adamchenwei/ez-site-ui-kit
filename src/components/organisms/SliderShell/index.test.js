import React from 'react';
import renderer from 'react-test-renderer';
import SliderShell from './index';
import { COMPONENT_LABEL, ES6 } from '../../../storybookonly/config/MENU_CATEGORIES';

describe(`${COMPONENT_LABEL}/${ES6}/SliderShell/Unit Test`, () => {
  test('renders as expected', () => {
    const tree = renderer.create(<SliderShell />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
