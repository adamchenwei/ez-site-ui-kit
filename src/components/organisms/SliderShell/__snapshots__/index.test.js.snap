// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component/ES6/SliderShell/Unit Test renders as expected 1`] = `
<section
  className="Container-gZqETv hbYoHa"
  style={Object {}}
/>
`;
